mainframer.sh

```diff
@@ -70,6 +74,36 @@ if [ -z "$REMOTE_COMMAND" ]; then
 	exit 1
 fi
 
+function convertSecondsToTime {
+    ((h=${1}/3600))
+    ((m=(${1}%3600)/60))
+    ((s=${1}%60))
+
+    #Find out the time format and correct string.
+    if [ "$h" -eq "1" ]; then
+        HOURS_TAG="hour"
+    fi
+    if [ "$m" -eq "1" ]; then
+        MINUTES_TAG="minute"
+    fi
+    if [ "$s" -eq "1" ]; then
+        SECONDS_TAG="second"
+    fi
+
+     #See if we can emit one of the time units, because it's 0.
+     if [ "$h" -eq "0" ] && [ "$m" -eq "0" ]; then
+        printf "%d $SECONDS_TAG\n" ${s}
+     elif [ "$h" -eq "0" ]; then
+        printf "%d $MINUTES_TAG %d $SECONDS_TAG\n" ${m} ${s}
+     elif [ "$m" -eq "0" ]; then
+        printf "%d $HOURS_TAG %d $SECONDS_TAG\n" ${h} ${s}
+     elif [ "$s" -eq "0" ]; then
+        printf "%d $HOURS_TAG %d $MINUTES_TAG\n" ${h} ${m}
+     else
+        printf "%d $HOURS_TAG %d $MINUTES_TAG %d $SECONDS_TAG\n" ${h} ${m} ${s}
+      fi
+}
```

Comment:
Pretty sure it is possible via [shorter and better approaches](http://unix.stackexchange.com/a/27014).